config/configauth/README.md

```diff
@@ -1,17 +1,27 @@
 # Authentication configuration for receivers
 
-This module allows server types, such as gRPC and HTTP, to be configured to perform authentication for requests and/or RPCs. Each server type is responsible for getting the request/RPC metadata and passing down to the authenticator. Currently, only bearer token authentication is supported, although the module is ready to accept new authenticators.
+This module allows server types, such as gRPC and HTTP, to be configured to perform authentication for requests and/or RPCs. Each server type is responsible for getting the request/RPC metadata and passing down to the authenticator.
 
-Examples:
+Authenticators are set up as extensions and passed to the server settings for a receiver. When writing a custom authenticator extension, it must call `configauth.AddAuthenticatorToRegistry` from within its createExtension factory method. This allows the receiver to look up the instance of the authenticator that it is configured to use. See [the OIDC extension factory](../../extension/oidcextension/factory.go) as an example.
```

Comment:
Global registries with self-registration are often sources of problems, make testing more difficult, etc. What if I want to have a test with 2 different instances of the extension? (Think of a test which creates 2 full instances of the Collector to test communication between 2 Collectors, which are perhaps configured with different sets of extensions).

A possible alternate is that the caller of createExtension probes the returned value to see if it implements the Authenticator interface and adds it to the Service's local registry of authenticators. We will need to find the way to make this local registry available at the point where the authenticators are created (in  `GRPCServerSettings.ToServerOption` function), not sure what exactly is the best way to do it (possibly add to the ReceiverCreateParams).

Reply:
> Think of a test which creates 2 full instances of the Collector to test communication between 2 Collectors, which are perhaps configured with different sets of extensions

To make sure I understood this case: this would be with a custom main, where I would create two [`Application`](https://pkg.go.dev/go.opentelemetry.io/collector/service#New) instances as part of the same binary, right? If so, I have to admit that I haven't thought about this case.

I think @bogdandrutu mentioned that he wanted to work on a view registry for the metrics views as well. For Jaeger, we've been thinking about using something similar to this PR here to allow people to extend Jaeger with their own storage mechanisms, where the storage would be a custom `extension` that people can build and register with an extension/component of ours. 

There's definitely a need for registries :-)

Reply:
>To make sure I understood this case: this would be with a custom main, where I would create two Application instances as part of the same binary, right? If so, I have to admit that I haven't thought about this case.

Yes.

>There's definitely a need for registries :-)

Yes, and registries in global variables are undesirable. See for example how we avoided global factory registry although there can be only one per Application. So whatever data we believe is a single instance per Collector needs to be in the Application, not a global variable.

Reply:
I'll try to scratch something as part of another PR, unless someone wants to work on it first.

Reply:
I can take a look at it as a part of this PR. 

I think I see two code paths forward right now, that I want to run past y'all since I haven't spent too much time inside the service code.

1. Have the extensions builder would also have to return this registry of Authenticators back up to the service.Application, which can then be passed into the pipeline setup code for the receivers to have access to.
2. Instead of an explicit registry for Authenticators, we can just pass the service's builtExtensions map to the receivers that can expect to find the authenticator as an extension

Let me know if either of those sound right, or if I'm completely off the mark here.

Reply:
> I can take a look at it as a part of this PR.

I'd really rather have it as part of another effort, as we need a proper solution for registries in general.

Reply:
@jpkrohling ok this sounds like it might be a bit more sweeping of a change then what I suggested above

Reply:
Can this PR move forward without this wider change to registries or does it need to wait until then?